test/test_linalg.py

```diff
@@ -4043,11 +4043,9 @@ def _test_linalg_solve_triangular(self, A, B, upper, left, uni):
         self.assertEqual(X, out)
 
     @dtypes(*floating_and_complex_types())
-    @precisionOverride({torch.float32: 1e-1, torch.complex64: 1e-1,
+    @precisionOverride({torch.float32: 2e-1 if TEST_WITH_ROCM else 1e-1, torch.complex64: 1e-1,
                         torch.float64: 1e-8, torch.complex128: 1e-8})
```

Comment:
Hmm, I wonder whether we could just modify inputs to be better conditioned? `1e-1` seems very low to me...

Reply:
@alugorey , `_test_linalg_solve_triangular` has a `well_conditioned` argument which is `False` by default. Could you please try this one instead and reduce the tolerance level. `1e-1` is just too low to be able to tell the function is doing the right thing.

Reply:
@nikitaved setting well_conditioned to true allows the test to pass. pushing up that change now

Reply:
@nikitaved I was able to bring it down to 1e-4 and still have it pass. 1e-4 fails. Would you like me to peg it there?

Reply:
`1e-4` seems fine, it is both for `float` and `cfloat`, right?

Reply:
1e-4 for float32, i was able to push complex64 to 1e-8 and still pass

Reply:
Let's just remove explicit tolerance for cfloat then!

Reply:
Done! removed for both complex types, both pass!

Reply:
@nikitaved uh oh! looks like 1e-4 is too tight for CPU test cases which are now failing

Reply:
Could you please just leave a comment on the current choose and that it is dictated by the CPU path?